File: JavaScript/programmers_045.js
function primeCheck(n){
    for(var i = 2; i <= Math.sqrt(n); i++){
        if(n % i == 0) {
            return false;
        }
    }
    return true;    
}

function solution(nums) {
    var answer = 0;

    const getComb = (arr, select) => {
        const result = []
        if(select === 1) return arr.map(item => [item]);

        arr.forEach((fixer, index, origin) => {
            const rest = origin.slice(index + 1);
            const comb = getComb(rest, select - 1);
            const atteched = comb.map((item) => [fixer, ...item]);
            result.push(...atteched);
        })
        return result;
    }

    var combNum = getComb(nums, 3)

    combNum.forEach((item) => {
        if(primeCheck(item.reduce((acc, cur) => {return acc + cur}, 0))) answer++;
    })

    return answer;
}

File: Python/Practice/programmers_045.py
def solution(n):
    answer = []
    board = [[0]*n for i in range(n)]
    x, y = -1, 0
    num = 1
    for i in range(n):
        for j in range(i, n):
            if i%3 == 0:
                x+=1
            if i%3 == 1:
                y+=1
            if i%3 == 2:
                x-=1
                y-=1
            board[x][y] = num
            num += 1
    for i in board:
        for j in i:
            if j != 0:
                answer.append(j)
    return answer